From cb3d761f37edcfa426d45e9c06d66b31a2c9fa3b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 16:28:52 +0800
Subject: [PATCH 2/2] hack(osgPlugins): drop FFmpeg module

Upstream has not been keeping up with FFmpeg updates and has not seen active
feature development for a while. Drop FFmpeg to save our own time (we still
have GStreamer).

Debian dropped dependency to FFmpeg, disabling this plugin implicitly. Fedora,
on the other hand, never enabled this plugin.

Do this more explicitly so we don't run into trouble if the build environment
came with FFmpeg installed.
---
 CMakeLists.txt                | 1 -
 src/osgPlugins/CMakeLists.txt | 4 ----
 2 files changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ef2edd33..a54a9c40b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -752,7 +752,6 @@ ELSE()
     FIND_PACKAGE(CURL)
     FIND_PACKAGE(LibVNCServer)
     FIND_PACKAGE(DCMTK)
-    FIND_PACKAGE(FFmpeg)
     FIND_PACKAGE(GStreamer COMPONENTS app pbutils)
     FIND_PACKAGE(GLIB COMPONENTS gobject)
     FIND_PACKAGE(DirectShow)
diff --git a/src/osgPlugins/CMakeLists.txt b/src/osgPlugins/CMakeLists.txt
index 812550412..d42865127 100644
--- a/src/osgPlugins/CMakeLists.txt
+++ b/src/osgPlugins/CMakeLists.txt
@@ -209,10 +209,6 @@ IF(OSG_CPP_EXCEPTIONS_AVAILABLE)
     ADD_PLUGIN_DIRECTORY(txp)
 ENDIF()
 
-IF(FFMPEG_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
-    ADD_PLUGIN_DIRECTORY(ffmpeg)
-ENDIF()
-
 IF(GSTREAMER_FOUND AND GLIB_FOUND)
     ADD_PLUGIN_DIRECTORY(gstreamer)
 ENDIF()
-- 
2.46.0

